iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.inl

```diff
@@ -189,7 +219,13 @@ inline void storable_function<Capacity, signature<ReturnType, Args...>>::move(st
     auto ptr = dest.m_storage.template allocate<CallableType>();
     cxx::Expects(ptr != nullptr);
 
-    auto obj = static_cast<CallableType*>(src.m_callable);
+    // AXIVION Next Construct AutosarC++19_03-M5.2.8: type erasure - conversion to compatible type
+    const auto obj = static_cast<CallableType*>(src.m_callable);
+
+    // AXIVION Next Construct AutosarC++19_03-A5.3.2: obj is guaranteed not to be nullptr
```

Comment:
See above. A `nullptr` check should be added.

Reply:
@MatthiasKillat Bad design and mistakes to use a moved-from object can always occur, but this is not acceptable in a safety-critical software! I know this might have a performance impact, but as you can read in the following post even the C++ community (here LLVM) is moving in this direction: https://discourse.llvm.org/t/rfc-c-buffer-hardening/65734

But you are completely right, this is the perfect place for a `Expects` (later `IOX_ASSERT`). The user can then decide if to compile out the checks in release mode or keep them active (similar to what the LLVM folks are planning)-

Reply:
Safety critical software needs more careful design to prevent the mistakes in the first place but is no excuse for inefficient code. It is not safer just because of additional checks for conditions that cannot happen in regular use (such conditions always exist one way or another for moderately complex systems).

However, I am all for asserts to state intention/assumptions.